google/generativeai/types/retriever_types.py

```diff
@@ -207,8 +221,28 @@ def create_document(
         if client is None:
             client = get_default_retriever_client()
 
-        document = None
-        if valid_name(name):
+        # Handle the custom_metadata parameter
+        c_data = []
```

Comment:
if + for

if the list is empty, the for does nothing.
You can de-nest things using:

```
if custom_metadata is None:
  custom_metadata = []
  
for ...
```